src/install/dracut-install.c

```diff
@@ -1367,81 +1367,95 @@ static int install_all(int argc, char **argv)
         return r;
 }
 
-static int install_firmware_fullpath(const char *fwpath)
+static int install_firmware_fullpath(const char *fwdir, const char *fwname,
+                                     const char *suffix)
 {
-        const char *fw = fwpath;
-        _cleanup_free_ char *fwpath_compressed = NULL;
-        struct stat sb;
-        int ret;
-        if (stat(fwpath, &sb) != 0) {
-                _asprintf(&fwpath_compressed, "%s.zst", fwpath);
-                if (access(fwpath_compressed, F_OK) != 0) {
-                        strcpy(fwpath_compressed + strlen(fwpath) + 1, "xz");
-                        if (access(fwpath_compressed, F_OK) != 0) {
-                                log_debug("stat(%s) != 0", fwpath);
-                                return 1;
-                        }
-                }
-                fw = fwpath_compressed;
+        _cleanup_globfree_ glob_t globbuf;
+        _cleanup_free_ char *fwpath_full;
+        int ok = 0, fail = 0;
+        const char *fw;
+        size_t i;
+
+        _asprintf(&fwpath_full, "%s/%s%s", fwdir, fwname, suffix);
+        glob(fwpath_full, 0, NULL, &globbuf);
+
+        if (globbuf.gl_pathc == 0) {
+                log_debug("firmware file '%s' not found", fwpath_full);
+                return -1;
         }
-        ret = dracut_install(fw, fw, false, false, true);
-        if (ret == 0) {
-                log_debug("dracut_install '%s' OK", fwpath);
+
+        for (i = 0; i < globbuf.gl_pathc; i++) {
+                fw = globbuf.gl_pathv[i];
+
+                if (access(fw, F_OK) != 0) {
+                        log_debug("stat(%s) != 0", fw);
+                        fail++;
+                        continue;
+                }
+
+                if (dracut_install(fw, fw, false, false, true) == 0) {
+                        log_debug("dracut_install '%s' OK", fw);
+                        ok++;
+                } else {
+                        log_debug("dracut_install '%s' FAIL", fw);
+                        fail++;
```

Comment:
i mean, realistically, this isn't an issue, but this *is* a very archaic style, and you *could*, nominally, overflow ok and fail all the way back to zero on platforms where sizeof(size_t) > sizeof(int); why not make them both bools?

Reply:
We don't need to know the count for anything, so yes, it makes more sense to use bools. I will change it.